lib/bundle.js

```diff
@@ -1,44 +1,37 @@
 var merge = require('merge-stream')
 var thr = require('through2')
-var buffer = require('vinyl-buffer')
-var path = require('path')
+var duplex = require('duplexify')
 
 module.exports = function (b, opts) {
-  var bundle = b.bundle.bind(b)
-  b.bundle = function () {
-    var out = buffer()
-    var stream = merge()
+  b.on('reset', reset)
+  reset()
 
-    function handleError(err) {
-      delete err.stream
-      out.emit('error', err)
-    }
+  function reset() {
+    var input = opts.toVinylStream(opts.common || 'common.js', opts.basedir)
+    var output = merge()
 
-    var common = bundle().on('error', handleError)
     if (opts.common || !opts.needFactor) {
-      b.emit('factor.pipeline', path.resolve(opts.basedir, opts.common || 'common.js'), b.pipeline)
-      stream.add(
-        common.pipe(
-          opts.toVinylStream(opts.common || 'common.js', opts.basedir)
-        ).on('error', handleError)
-      )
-    } else {
-      // consume `common`, so b.on('bundle', (bundle) => {}) will work
-      common.pipe(thr.obj())
+      output.add(input)
     }
+
     if (opts.needFactor) {
       var wait = thr.obj()
-      stream.add(wait)
+      output.add(wait)
       b.once('factor.outputs', function (streams) {
         streams.forEach(function (s) {
           s.on('error', handleError)
         })
-        stream.add(streams)
+        output.add(streams)
         wait.end()
       })
     }
 
-    return stream.pipe(out)
+    b.pipeline.get('wrap').push(duplex.obj(input, output))
```

Comment:
In my testing I added the watchify plugin just before factor-vinylify to avoid this issue. I'm not totally sure what the "right" thing to do here is, though I'd be happy to change it. I wonder if @substack has any thoughts about this.